exercises/main.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+)
+
+type Entry struct {
+	ID          uint32
+	FirstName   string
+	LastName    string
+	PhoneNumber string
+}
+
+func newEntry() {
+	const phonebookFile = "phonebook.json"
+	var entries []Entry
+	var ID uint32
+	e := Entry{ID: ID}
+	f, err := os.Open(phonebookFile)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+	if err := json.NewDecoder(bufio.NewReader(f)).Decode(&entries); err != nil {
+		log.Fatal(err)
+	}
+
+	scanner := bufio.NewScanner(os.Stdin)
```

Comment:
It is a good idea to create it once in`main` and then to pass it everywhere:
```go
func main() {
  s := bufio.NewScanner(os.Stdin)
  for {
    ...
    addEntry(s)
    ...
  }
}
```
This way you guarantee no glitches